src/js/lib/timezones.js

```diff
@@ -0,0 +1,102 @@
+import { get, set } from 'idb-keyval';
+
+/**
+ * Makes timezones awesome
+ */
+export class TimezoneChanger {
+  /**
+   * @param {DOMElement} changer - The timezone changer element.
+   */
+  constructor(changer) {
+    this.changer = changer;
+    this.times = Array.from(document.querySelectorAll('[data-time]')).map((el) => ({
+      el,
+      date: new Date(el.dataset.time),
+      format: JSON.parse(el.dataset.format),
+    }));
+    this.dates = Array.from(document.querySelectorAll('[data-date]')).map((el) => ({
+      el,
+      date: new Date(el.dataset.date),
+      format: JSON.parse(el.dataset.format),
+    }));
+    this.timezones = Array.from(document.querySelectorAll('[data-timezone]'));
+
+    if (changer) {
+      changer.addEventListener('change', this.changeTZ.bind(this));
+    }
+
+    this.init();
+  }
+
+  /**
+   * Gets the user's stored timezone
+   */
+  async init() {
+    const offset = new Date().getTimezoneOffset();
+    this.local = `GMT${offset > 0 ? '-' : '+'}${Math.floor(offset / 60)
+      .toString()
+      .padStart(2, 0)}:${Math.floor(offset % 60)
```

Comment:
Same as above for XX:30 timezones, like Mumbai

```suggestion
      .padStart(2, 0)}:${Math.abs(Math.floor(offset % 60))
```